File: EGovernment_backend/statistics-service/domain/request.go
package domain

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Request struct {
	ID          primitive.ObjectID `bson:"_id,omitempty" json:"id"`
	Name        string             `bson:"name" json:"name"`
	Lastname    string             `bson:"lastname" json:"lastname"`
	Email       string             `bson:"email" json:"email" validate:"required,email"`
	PhoneNumber int                `bson:"phone_number" json:"phone_number"`
	Category    CategoryPerson     `bson:"category" json:"category"`
	Question    string             `bson:"question" json:"question"`
	Date        primitive.DateTime `bson:"date" json:"date"`
}

type CategoryPerson string

const (
	Student     = "Student"
	Scientist   = "Scientist"
	Researcher  = "Researcher"
	Analyst     = "Analyst"
	PrivateUser = "PrivateUser"
	Other       = "Other"
)

File: EGovernment_frontend/src/app/models/statisics/request.ts
import {CategoryPerson} from "./categoryPerson";

export class Request {
  id: string;
  name: string;
  lastname: string;
  email: string;
  phone_number: number;
  category: CategoryPerson;
  question: string;
  date: string;

  constructor(id: string, name: string, lastname: string, email: string, phone_number: number, category: CategoryPerson, question: string, date: string) {
    this.id = id;
    this.name = name;
    this.lastname = lastname;
    this.email = email;
    this.phone_number = phone_number;
    this.category = category;
    this.question = question;
    this.date = date;
  }
}
